### PR TITLE
Create a single artifact for the server module

### PR DESCRIPTION
**Pull request by vikrantpuppala**
**PR Checklist**

- [ ] A description of the changes is added to the description of this PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

Create single artifact for the server module

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -385,13 +385,15 @@ lazy val server = (project in file("server"))
       val log = streams.value.log
       (Test / runMain).toTask(s" io.unitycatalog.server.utils.PopulateTestDatabase").value
     },
-    Test / javaOptions += s"-Duser.dir=${(ThisBuild / baseDirectory).value.getAbsolutePath}"
+    Test / javaOptions += s"-Duser.dir=${(ThisBuild / baseDirectory).value.getAbsolutePath}",
+    Compile / packageBin / mappings ++= (Compile / packageBin / mappings).value ++
```
Comment (tdas) [inline build.sbt:389]: add a comment explaining what you are doing.

File: build.sbt [hunk 2]
```diff
@@ -67,7 +66,9 @@ lazy val commonSettings = Seq(
     val packageFile = (Compile / packageBin).value
     generateClasspathFile(
       targetDir = packageFile.getParentFile,
-      classpath = (Runtime / dependencyClasspath).value)
+      // Also include the jar being built (packageFile) in the classpath
+      classpath = (Runtime / dependencyClasspath).value :+ Attributed.blank(packageFile)
```
Comment (tdas) [inline build.sbt:70]: add comment saying that this is needed for the controlModels and serverModel projects since they are marked as provided.

**Review by tdas — APPROVED** [review]

**Comment by tdas** [comment]
thank you @vikrantpuppala for fixing this.

**Comment by vikrantpuppala** [comment]
@tdas addressed your comment on this, could you please merge

**Pull request merged**